tests/system/test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py

```diff
@@ -0,0 +1,197 @@
+                                                       f'cut -c 1-3 {WAZUH_PATH}/etc/client.keys'), host_manager)
+    clean_cluster_logs(testinfra_hosts, host_manager)
+
+
+@pytest.mark.parametrize('test_case', [cases for cases in test_cases_yaml], ids=[cases['name']
+                         for cases in test_cases_yaml])
+def test_correct_merged_file_generation(test_case, environment_setting):
+    '''
+        description: Checking correct merged file generation.
+        wazuh_min_version: 4.5.0
+        parameters:
+            - test_case:
+                type: list
+                brief: List of tests to be performed.
+            - environment_setting:
+                type: function
+                brief: Clear files, directories and logs, reset initial conditions in /var/ossec/share
+                        (includes agent enrollment).
+                        Also stops the manager if it is required.
+        assertions:
+            - check merged.mg in the selected folder and the created file.
+            - check if merged contains the correct information.
+            - check if log contains the proper information in case the added file has no data.
+        input_description: Different use cases are found in the test module and include parameters.
+        expected_output:
+            - merged.mg should be created and modified automatically considering the file/s and its/their information.
+    '''
+    # Declaring variables
+    metadata = test_case['metadata']
+    action = metadata['action']
+    file_content = metadata['file_content']
+    number_files = metadata['number_files']
+    test_type = metadata['test_type']
+    folder = metadata['shared_folder']
+    file_name = metadata['file_name']
+    files_list = []
+
+    # Main action of the test
+    if action == "remove":
+        host_manager.run_command(testinfra_hosts[0], f'rm {WAZUH_PATH}/etc/shared/default/merged.mg -f')
+    elif action == "add_files":
+        if number_files == 1:
+            host_manager.run_command(testinfra_hosts[0], f"touch {WAZUH_PATH}/etc/shared/{folder}/{file_name}.txt")
+            if file_content != 'zero':
+                host_manager.modify_file_content(host=testinfra_hosts[0],
+                                                 path=f"{WAZUH_PATH}/etc/shared/{folder}/{file_name}.txt",
+                                                 content=file_content)
+        else:
+            for number in range(number_files):
+                files_list.append(f'{file_name}{number + 3}')
+            for file in files_list:
+                host_manager.run_command(testinfra_hosts[0], f"touch {WAZUH_PATH}/etc/shared/{folder}/{file}.txt")
+                if file_content != 'zero':
+                    host_manager.modify_file_content(host=testinfra_hosts[0],
+                                                     path=f"{WAZUH_PATH}/etc/shared/{folder}/{file}.txt",
+                                                     content=file_content)
```

Comment:
Why is the case of 1 file separated from the case of several files?

Reply:
Changes done in https://github.com/wazuh/wazuh-qa/pull/4129/commits/d7a26fade0a1f1ffdb7bc46d88c1c8f0b2fedb38